src/analytics/internal/analytics.js

```diff
@@ -0,0 +1,153 @@
+import { MapEventTypeToInt, MapNotifTypeToIntArray } from './constants'
+import SendToServer from './sendToServer'
+import { remoteFunction } from 'src/util/webextensionRPC'
+import AnalyticsLogStorage from './AnalyticsLogStorage'
+
+const getCount = remoteFunction('getCount')
+const getLatestEvent = remoteFunction('getLatestEvent')
+
+class Analytics {
+    _operationsMap = {
+        successful_search: undefined,
+        unsuccessful_search: undefined,
+        datepicker: undefined,
+        bookmark_filter: undefined,
+        tag_filter: undefined,
+        domain_filter: undefined,
+        tagging: undefined,
+        bookmark: undefined,
+        blacklist: undefined,
+        address_bar_search: undefined,
+        datepicker_nlp: undefined,
+        nlp_search: undefined,
+    }
+
+    async registerOperations() {
```

Comment:
I think this class really needs a Promise property which you can set and resolve in this, `registerOperations`, method. This and `loadInitialData` are async stuff so they're going to take some amount of time before they finish, and in that time an event could be logged, updating your `_operationsMap` property and possibly breaking `loadInitialData`. 

All you need to do is add a Promise property, maybe `this._initializationPromise` which gets set to a resolved Promise once `registerOperations` completes. Then each of your methods that relate to event logging and mutate `this._operationsMap` just need to add a single `await this._initializationPromise` as their first line. This will ensure event log requests that happen while `registerOperations` is still running will wait for it to finish before running.

There's a bit of an example with a similar thing here (note all the public methods `await this._initializationPromise`): https://github.com/WorldBrain/Memex/blob/develop/src/search/search-index-new/storage/manager.ts#L21